docs/DeveloperGuide.md

```diff
@@ -1,14 +1,123 @@
+As such, these will be passed in directly to their **`Command`** classes without a **`argumentType`** pair.
+
+More complex command words such as **`add`** and **`update`** necessitated multiple arguments.
+To implement this while ensuring that multi-word strings are acceptable input, **`extractArguments`** was implemented. 
+Without specifically explaining the main regular expression ([details here](https://regex101.com/r/gwjHWD/3)), the approach sought to match **`argumentTag/argumentValue`** pairs with a positive lookahead. 
+The final argument pair will then be extracted using a separate regex. 
+Together, this ensured that all argument pairs can be effectively parsed and dispatched to each **`Command`** class.
+
+Throughout the `Parser` implementation, exceptions were used to return `IncorrectCommand` classes that can be used to pass error messages to the user. These will be discussed in the following segments.
+
+### Update feature
+
+#### Current Design and Implementation
+
+The update feature is facilitated by `UpdateCommand`. It extends `ModificationCommand` and implements the following operations:
+
+* `UpdateCommand#generateUpdatePairs()` — Generates pairs of attributes and their update values.
+* `VersionedAddressBook#generateUpdateString()` — Generates sString with details of the update executed.
+
+Given below is an example usage scenario and how the update feature behaves at each step.
+
+Step 1. The user adds an equipment to the system with the help of the `add` command. The added equipment has the following attributes shown below.
+
+![equipment0](images/equipment0.png)
+
+Step 2. The user executes `update s/S1404115ASF n/SpeakerC c/2000 pd/2022-01-29` to update equipment with serial number S1404115ASF. `Parser#parseCommand` is called from `Duke` to parse the user's input.
+
+Step 3. The parser recognises that an `UpdateCommand` is required, and the UpdateCommand is prepared to return to `Duke`. In the constructor of `UpdateCommand`, `prepareModification` is called to set the values of the attributes to be updated. The other attributes are set to null by default.
+
+Step 4. `UpdateCommand#execute` is run to process the update. If the serialNumber attribute is null, a `CommandResult` with a `MISSING_SERIAL_NUMBER` output string will be returned. Otherwise, `EquipmentManager#updateEquipment` is called.
+
+Step 5. If the update was successful, a `CommandResult` with success message will be returned, else a `CommandResult` with `UPDATE_FAILURE_MESSAGE` will be returned. Upon successful update, the object should be updated with the new attributes as shown in the diagram below.
+
+![equipment1](images/equipment1.png)
+
+The *Sequence Diagram* below shows how the objects/classes interact with each other in the case of an update command being issued by the user.
+
+![sequenceDiagram](images/sequenceDiagram.png)
```

Comment:
![image](https://user-images.githubusercontent.com/60244374/161184109-0c93d1ce-d68e-437d-b389-03f7d33ec380.png)

Is Parser making a new command object here? If Parser is making a new command object, it may be the diagram should signify a constructor call (by having the calling arrow touch the "command: UpdateCommand" box and have an activation bar immediately following that.